### PR TITLE
[backport-14194-stabe8] URLEncode logout attribute

### PR DESCRIPTION
**Pull request by LukasReschke**
Otherwise logout can fail if the requesttoken contains a +

### CODE REVIEW

**Comment by LukasReschke** [comment]
:gun:

**Comment by LukasReschke** [comment]
@owncloud-bot Retest this please.

**Comment by DeepDiver1975** [comment]
> :gun:

try that :bomb:

**Comment by LukasReschke** [comment]
Seems to be more effective – yes.

**Comment by LukasReschke** [comment]
For stable8 I changed the RNG to the previous charset since there are some apps that do URL generation the wrong way (such as gallery *sigh*). See https://github.com/owncloud/core/issues/14228 for the general issue *sigh*

File: lib/private/security/securerandom.php [hunk 1]
```diff
@@ -64,12 +64,11 @@ public function getMediumStrengthGenerator() {
 	 * Generate a random string of specified length.
 	 * @param string $length The length of the generated string
 	 * @param string $characters An optional list of characters to use if no characterlist is
-	 * 							specified 0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ./
-	 * 							is used.
+	 * 							specified all valid base64 characters except+ are used.
```
Comment (nickvergessen) [inline lib/private/security/securerandom.php]: I'd expand that `specified all valid base64 characters except + (plus sign) are used.`
Also space missing after except

**Comment by nickvergessen** [comment]
I wonder if we should have a "URL save" letter group, that can be used for this.

**Comment by LukasReschke** [comment]
> I wonder if we should have a "URL save" letter group, that can be used for this.

Mhm… Something for master then. But can anyways already achieved by combining the existing flags.

File: lib/private/security/securerandom.php (hunk 1 above)
Comment (LukasReschke) [inline lib/private/security/securerandom.php]: ACK

**Comment by nickvergessen** [comment]
:+1:

**Comment by PVince81** [comment]
Makes sense, logout still works :+1: 

@karlitschek please approve this backport

**Comment by blizzz** [comment]
I cannot log out on stable8 without this PR … recommend backporting :D :+1:

**Comment by karlitschek** [comment]
backport is fine with me.

**Pull request merged**